webapi/src/main/scala/org/knora/webapi/responders/v1/OntologyResponderV1.scala

```diff
@@ -462,4 +498,240 @@ class OntologyResponderV1 extends ResponderV1 {
+                propertyTypesInNamedGraphResponse <- (storeManager ? SparqlSelectRequest(propertyTypesInNamedGraphSparql)).mapTo[SparqlSelectResponse]
+
+                propertyTypeIrisInNamedGraph = propertyTypesInNamedGraphResponse.results.bindings.map {
+                    (row) => row.rowMap("property")
+                }.toVector
+
+
+                // cache these results
+                namedGraphEntity = NamedGraphEntityInfoV1(
+                    namedGraphIri = namedGraphIriToQuery,
+                    propertyIris = propertyTypeIrisInNamedGraph,
+                    resourceClasses = resourceClassIrisInNamedGraph
+                )
+            } yield namedGraphEntity
+        }
+
+        for {
+
+            cacheKeyForNamedGraph <- Future(new OntologyCacheKeys.NamedGraphInfoKey(namedGraphIri))
+
+            namedGraphCacheResult <- CacheUtil.getOrCacheItem(
+                cacheName = OntologyCacheName,
+                cacheKey = cacheKeyForNamedGraph,
+                queryFun = queryNamedGraphEntityV1
+            )
+        } yield namedGraphCacheResult
+
+
+    }
+
+    /**
+      * Gets all the resource classes and their properties for a named graph.
+      *
+      * @param namedGraphIriOption the Iri of the named graph or None if all the named graphs should be queried.
+      * @param userProfile         the profile of the user making the request.
+      * @return [[ResourceTypesForNamedGraphResponseV1]].
+      */
+    private def getResourceTypesForNamedGraph(namedGraphIriOption: Option[IRI], userProfile: UserProfileV1): Future[ResourceTypesForNamedGraphResponseV1] = {
+
+        // get the resource types for a named graph
+        def getResourceTypes(namedGraphIri: IRI): Future[Vector[ResourceTypeV1]] = {
+            for {
+
+            // get NamedGraphEntityInfoV1 for the given named graph
+                namedGraphEntityInfo: NamedGraphEntityInfoV1 <- getNamedGraphEntityInfoV1ForNamedGraph(namedGraphIri, userProfile)
+
+                // get resinfo for each resource class in namedGraphEntityInfo
+                resInfosForNamedGraphFuture: Seq[Future[(String, ResourceTypeResponseV1)]] = namedGraphEntityInfo.resourceClasses.map {
+                    (resClassIri) =>
+                        for {
+                            resInfo <- getResourceTypeResponseV1(resClassIri, userProfile)
+                        } yield (resClassIri, resInfo)
+                }
+
+                resInfosForNamedGraph <- Future.sequence(resInfosForNamedGraphFuture)
+
+                resourceTypes: Vector[ResourceTypeV1] = resInfosForNamedGraph.map {
+                    case (resClassIri, resInfo) =>
+
+                        val properties = resInfo.restype_info.properties.map {
+                            (prop) => PropertyTypeV1(
+                                id = prop.id,
+                                label = prop.label.getOrElse("")
```

Comment:
I think a `label` should not be optional, but required in the ontology. Maybe we should better throw an error if label is not given for a resource class or a property.